src/app/folder/folderlist.component.ts

```diff
@@ -220,6 +277,120 @@ export class FolderListComponent {
         ).subscribe(() => this.messagelistservice.refreshFolderCount());
     }
 
+    async folderReorderingDrop(sourceFolderId: number, destinationFolderId: number, aboveOrBelowOrInside: number) {
+        if (sourceFolderId === destinationFolderId) {
+            // can't move a folder above, below or inside itself
+            return;
+        }
+
+        const folders = await this.messagelistservice.folderCountSubject.pipe(take(1)).toPromise();
+        let sourceIndex = folders.findIndex(fld => fld.folderId === sourceFolderId);
+        let destinationIndex = folders.findIndex(folder => folder.folderId === destinationFolderId);
+
+        console.log(`move folder ${folders[sourceIndex].folderPath} ${
+                            aboveOrBelowOrInside === 1 ? 'above' :
```

Comment:
We don't seem to be using the enum names here, is there a reason?

Reply:
Thanks! I missed that one.